### PR TITLE
Join directory

### PR DESCRIPTION
**Pull request by ChowdhuryShaswata**
The directory end of the join functionality in the text editor has been implemented.

JoinController: An Activity that manages and runs the file selection screen for Join. Send this the list of files currently selected to join together.

JoinListAdapter: A custom adapter for the list of displayed files. Extends SelectionListAdapter as much of the functionality is similar. Overrides one method to change the file menu controller used on a long click.

JoinMenuController: extends FileMenuController. Changes open file functionality to match the join process alongside JoinMenuInteractor.

### CODE REVIEW

**Comment by CSEndeavour** [comment]
Remember to add documentation.

**Review by CSEndeavour — APPROVED** [review]

**Comment by CSEndeavour** [comment]
Remove the login placeholder.

**Review by CSEndeavour — APPROVED** [review]

File: app/src/main/java/com/example/notes2text/directoryJoin/usecases/JoinMenuInteractor.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import com.example.notes2text.adapters.ActivitySwitchController;
+import com.example.notes2text.directoryJoin.adapters.JoinController;
+import com.example.notes2text.usecases.FileMenuInteractor;
+
+import java.io.File;
+import java.util.ArrayList;
+
+public class JoinMenuInteractor extends FileMenuInteractor {
+
+    ArrayList<File> selectedList;
```
Comment (JerryLi-617) [inline app/src/main/java/com/example/notes2text/directoryJoin/usecases/JoinMenuInteractor.java:19]: consider making this variable private

**Review by JerryLi-617 — APPROVED** [review]

File: app/src/main/java/com/example/notes2text/usecases/FileOpenInteractor.java [hunk 2]
```diff
@@ -42,8 +42,14 @@ public void openFile(Context context, File fileToOpen){
                 Bundle bundle = new Bundle();
                 bundle.putSerializable("file", fileToOpen);
                 docIntent.putExtras(bundle);
-                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```
Comment (julianmarrero) [inline app/src/main/java/com/example/notes2text/usecases/FileOpenInteractor.java:45]: Should be docIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)

**Review by julianmarrero — APPROVED** [review]

File: app/src/main/java/com/example/notes2text/usecases/FileOpenInteractor.java (hunk 2 above)
Comment (ChowdhuryShaswata) [reply app/src/main/java/com/example/notes2text/usecases/FileOpenInteractor.java:45]: Fixed the issue, thanks.

File: app/src/main/java/com/example/notes2text/directoryJoin/usecases/JoinMenuInteractor.java (hunk 1 above)
Comment (ChowdhuryShaswata) [reply app/src/main/java/com/example/notes2text/directoryJoin/usecases/JoinMenuInteractor.java:19]: should be private by default, but will specify in pull request alongside documentation.

**Pull request merged**